util/topgen/intermodule.py

```diff
@@ -390,6 +387,54 @@ def check_intermodule_field(obj: OrderedDict, prefix: str = "") -> int:
     return error
 
 
+def find_otherside_modules(topcfg: OrderedDict, m,
+                           s) -> List[Tuple[str, str, str]]:
+    """Find far-end port based on given module and signal name
+    """
+    signame = "{}.{}".format(m, s)
+    for req, rsps in topcfg["inter_module"]["connect"].items():
+        if req.startswith(signame):
+            # return rsps after splitting module instance name and the port
+            result = []
+            for rsp in rsps:
+                rsp_m, rsp_s, rsp_i = filter_index(rsp)
+                result.append(('connect', rsp_m, rsp_s))
+            return result
+
+        for rsp in rsps:
+            if signame == rsp:
+                req_m, req_s, req_i = filter_index(req)
+                return [('connect', req_m, req_s)]
+
+    # If reaches here, no matching results in 'connect'
+    # so search 'top' or 'external'
+
+    # todo: something here
+    for sig in topcfg["inter_signal"]["signals"]:
+        # Special Handling for custom connect
+        if m == "main" and s == "tl_corei":
+            return [('top', 'rv_core_ibex', 'tl_i')]
+
+        if m == "main" and s == "tl_cored":
+            return [('top', 'rv_core_ibex', 'tl_d')]
+
+        if m == "main" and s == "tl_dm_sba":
+            return [('top', 'dm_top', 'tl_h')]
+
+        if m == "main" and s == "tl_debug_mem":
+            return [('top', 'dm_top', 'tl_d')]
+
+        # If not falls into the special category, no way to find other side
+
+        if sig["inst_name"] == m and sig["name"] == s:
+            return [('top', '', sig["top_signame"])]
```

Comment:
I think you could probably write this more simply as something like this (untested):
```python
    req_pairs = {
        ('main', 'tl_corei'): ('rv_core_ibex', 'tl_i'),
        ('main', 'tl_cored'): ('rv_core_ibex', 'tl_d'),
        ('main', 'tl_dm_sba'): ('dm_top', 'tl_h'),
        ('main', 'tl_debug_mem'): ('dm_top', 'tl_d')
    }
    for sig in topcfg["inter_signal"]["signals"]:
        req_pairs[(sig['inst_name'], sig['name'])] = ('', sig['top_signame'])

    req_pair = req_pairs.get((m, s))
    if req_pair is not None:
        return [('top', req_pair[0], req_pair[1])]
```

It's a bit shorter and the "data" is more clearly separated from the logic.

Reply:
Always learn from your comments. Thanks!